app/models/query_custom_field_column.rb

```diff
@@ -46,8 +48,24 @@ def custom_field
     @cf
   end
 
-  def value(issue)
-    cv = issue.custom_values.detect { |v| v.custom_field_id == @cf.id }
+  def value(work_package)
+    cv = work_package.custom_values.detect { |value| value.custom_field_id == @cf.id }
     cv && cv.typed_value
   end
+
+  def sum_of(work_packages)
+    if work_packages.respond_to?(:joins)
+      # we can't perform the aggregation on the SQL side. Try to filter useless rows to reduce work.
+      cv_alias = "joined_#{name}"
+      work_packages = work_packages
+                      .joins("LEFT OUTER JOIN #{CustomValue.table_name} #{cv_alias} ON
+                              #{cv_alias}.customized_type='WorkPackage' AND
+                              #{cv_alias}.customized_id=#{WorkPackage.table_name}.id AND
+                              #{cv_alias}.custom_field_id=#{@cf.id}")
+                      .where("#{cv_alias}.value IS NOT NULL AND #{cv_alias}.value != ''")
```

Comment:
I can use it to perform the join, as I don't even need the **outer** join (my `where` will only accept inner join results anyway).

However, the `where` will remain as string, since I believe it is not possible to do these negated statements using AR hashes...